Code/Framework/AzCore/AzCore/Memory/AllocatorManager.cpp

```diff
@@ -14,394 +14,710 @@
+                            address,
+                            info.m_byteSize,
+                            info.m_alignment)
+                        .c_str());
+                }
+                else
+                {
+                    tracer.Output(
+                        MemoryTag,
+                        AllocationString::format("Allocation Addr: 0%p Size: %d Alignment: %d\n", address, info.m_byteSize, info.m_alignment).c_str());
+                }
+
+                if (includeAllocationLineAndCallstack)
+                {
+                    // If there is no stack frame records, output the location where the allocation took place
+                    if (!info.m_stackFrames)
+                    {
+                        // For the callstack use the NoWindow constant to have the stack frames be visually aligned beneath the previous
+                        // "mem" window entry
+                        tracer.Output(
+                            AZ::Debug::Trace::GetNoWindow(),
+                            AllocationString::format(
+                                R"( "%s" (%d))"
+                                "\n",
+                                info.m_fileName,
+                                info.m_lineNum)
+                                .c_str());
+                    }
+                    else
+                    {
+                        constexpr unsigned int MaxStackFramesToDecode = 30;
+                        Debug::SymbolStorage::StackLine decodedStackLines[MaxStackFramesToDecode];
+
+                        for (size_t currentStackFrame{}; numStackLevels > 0;)
+                        {
+                            // Decode up to lower of MaxStackFramesToDecode and the remaining number of stack frames per loop iteration
+                            unsigned int stackFramesToDecode = AZStd::GetMin(MaxStackFramesToDecode, static_cast<unsigned int>(numStackLevels));
+                            Debug::SymbolStorage::DecodeFrames(info.m_stackFrames + currentStackFrame, stackFramesToDecode, decodedStackLines);
+                            for (unsigned int i = 0; i < stackFramesToDecode; ++i)
+                            {
+                                if (info.m_stackFrames[currentStackFrame + i].IsValid())
+                                {
+                                    tracer.Output(AZ::Debug::Trace::GetNoWindow(), AllocationString::format(" %s\n", decodedStackLines[i]).c_str());
+                                }
+                            }
+                            numStackLevels -= stackFramesToDecode;
+                            currentStackFrame += stackFramesToDecode;
+                        }
+                    }
+                }
+            }
+
+            ++allocationCount;
+
+            return true;
+        };
+
+        AZ::Debug::Trace::Instance().Output(
+            MemoryTag, AllocationString("Dumping allocation records. This may take awhile\n").c_str());
```

Comment:
Bonus points if we can provide an estimated number of minutes based on the current number of allocations. ;)

```suggestion
        AZ::Debug::Trace::Instance().Output(
            MemoryTag, AllocationString("Dumping allocation records. This may take several minutes\n").c_str());
```

Reply:
Yeah I can try to figure out a good estimate for long it would take to dump out the records.